src/libcore/fmt/mod.rs

```diff
@@ -844,12 +844,36 @@ impl Display for char {
     }
 }
 
+// Two extra bytes for 0x
+#[cfg(target_pointer_width = "32")]
+const POINTER_PADDING: Option<usize> = Some(10);
+#[cfg(target_pointer_width = "64")]
+const POINTER_PADDING: Option<usize> = Some(18);
```

Comment:
Sure. I guess the only correctness issue with it is that right now, if you build on a platform that doesn't have 32 or 64 bit wide pointers, it will explode violently.

Granted, there's no obvious sign of anyone trying that soon, but the most obvious `cfg!` based construction will silently skip a function call on such a platform. How defensive is it reaonable to be?